puzzlejs/puzzle.js

```diff
@@ -985,19 +1004,22 @@ function PuzzleEntry(p, index) {
             if (otherTo) { this.undoManager.addUnit(toGrid, cellTo, otherAttributeName, parseInt(otherTo), parseInt(otherTo) & ~directionTo); }
             return true;
         }
-        else if (((codeFrom & directionFrom) && (codeTo & directionTo)) || ((otherFrom & directionFrom) && (otherTo & directionTo))) {
+        else if ((codeFrom & directionFrom) && (codeTo & directionTo)) {
             var givenAttributeName = "data-given-" + attributeNameBase + "-code";
             var givenCodeFrom = cellFrom.getAttribute(givenAttributeName);
             var givenCodeTo = cellTo.getAttribute(givenAttributeName);
             if (!givenCodeFrom) { givenCodeFrom = 0; } else { givenCodeFrom = parseInt(givenCodeFrom); }
             if (!givenCodeTo) { givenCodeTo = 0; } else { givenCodeTo = parseInt(givenCodeTo); }
             if (!(givenCodeFrom & directionFrom) && !(givenCodeTo & directionTo)) {
-                if (codeFrom) { this.undoManager.addUnit(fromGrid, cellFrom, attributeName, codeFrom, codeFrom & ~directionFrom); }
-                if (codeTo) { this.undoManager.addUnit(toGrid, cellTo, attributeName, codeTo, codeTo & ~directionTo); }
-                if (otherFrom) { this.undoManager.addUnit(fromGrid, cellFrom, otherAttributeName, otherFrom, otherFrom & ~directionFrom); }
-                if (otherTo) { this.undoManager.addUnit(toGrid, cellTo, otherAttributeName, otherTo, otherTo & ~directionTo); }
-
-                if ((attributeNameBase === "spoke") && (parseInt(this.options["data-spoke-levels"]) === 2) && ((codeFrom) && (codeTo))) {
+                this.undoManager.addUnit(fromGrid, cellFrom, attributeName, codeFrom, codeFrom & ~directionFrom);
+                this.undoManager.addUnit(toGrid, cellTo, attributeName, codeTo, codeTo & ~directionTo);
+
+                if ((currentSpokeLevel > 0) && ((currentSpokeLevel + 1) < maxSpokeLevels)) {
```

Comment:
Let's do an `attributeNameBase` check here too, just to be super safe. I know it isn't strictly required in this PR since `currentSpokeLevel > 0` can only be true when `attributeNameBase.includes("spoke")` but that might change somehow someday...